game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProTerritoryValueUtils.java

```diff
@@ -177,19 +176,31 @@ public static double findTerritoryAttackValue(
     return territoryValueMap;
   }
 
-  private static int findMaxLandMassSize(final GamePlayer player) {
-    int maxLandMassSize = 1;
+  protected static int findMaxLandMassSize(final GamePlayer player) {
     final GameData data = player.getData();
+    final Predicate<Territory> cond = ProMatches.territoryCanPotentiallyMoveLandUnits(player, data);
+    class LandMassFinder extends BreadthFirstSearch {
+      int landMassSize = 0;
+
+      LandMassFinder(final Territory startTerritory) {
+        super(startTerritory, cond);
+      }
+
+      @Override
+      public void visit(final Territory territory) {
+        landMassSize++;
+      }
+    }
+
+    int maxLandMassSize = 1;
+    final var fullVisited = new HashSet<Territory>();
     for (final Territory t : data.getMap().getTerritories()) {
-      if (!t.isWater()) {
-        final int landMassSize =
-            1
-                + data.getMap()
-                    .getNeighbors(
-                        t, 6, ProMatches.territoryCanPotentiallyMoveLandUnits(player, data))
-                    .size();
-        if (landMassSize > maxLandMassSize) {
-          maxLandMassSize = landMassSize;
+      if (!t.isWater() && !fullVisited.contains(t)) {
+        final LandMassFinder finder = new LandMassFinder(t);
+        finder.search();
+        fullVisited.addAll(finder.getVisited());
+        if (finder.landMassSize > maxLandMassSize) {
+          maxLandMassSize = finder.landMassSize;
```

Comment:
In java getters and setters are the preffered way of modifying fields, even on this small scale.
You can use lomboks `@Getter` and `@Setter` annotations if you want

Reply:
@RoiEXLab I'm not positive that is always the case. Within a class for example using getter/setter can be brittle as any child overrides can unintentionally change many methods. Essentially java was designed to expect inheritance as a default, then effective java came along and basically said to disallow it by default or else explicitly design for it (complex).

Reply:
In this case, it's accessing a what's essential a local variable. It can't be local to the method because anon or inner classes can't set primitives (i.e. can't have non-final variables). So in this case it's declared as a member of the inner class.

Another option is to declare an array or some other wrapper in the method itself as a local, e.g.

int maxLandMassSize[] = new int[1];

And have the inner class access maxLandMassSize[0], then it wouldn't need to be in the inner class (and the inner class could become anonymous).

But that seemed uglier, so I went with this.

By the way, if we structured the API without inheritance (e.g. the BFS class could take another interface instance to receive the new call backs), this doesn't solve this problem as the same problem exists. The extra object still needs to keep track of maxLandMassSize which is effectively something that's local to to the method and doesn't need a getter.